src/stores/ProfileController.ts

```diff
@@ -37,78 +24,118 @@ const profileSchema = yup.object().shape({
   }),
 });
 
-const isValidProfile = (profile: unknown): profile is ProfilesData => {
-  try {
-    profileSchema.validateSync(profile);
-    return true;
-  } catch (e: unknown) {
-    return false;
-  }
-};
+@injectable()
+export default class ProfileController {
+  private readonly profileService?: ProfileService;
 
-export const loadPersistedProfile = () => {
-  const profile = persist.getItem(PERSIST_PROFILE);
-  if (isValidProfile(profile)) {
-    useProfileStore.getState().setProfile(profile);
-    return profile;
+  constructor(@inject('INTEGRATION_TYPE') integrationType: keyof typeof INTEGRATION) {
+    this.profileService = getNamedModule(ProfileService, integrationType, false);
   }
-  useProfileStore.getState().setProfile(null);
-  return null;
-};
 
-export const initializeProfile = async ({ profile }: { profile: ProfilesData }) => {
-  persistProfile({ profile });
-  useFavoritesStore.setState({ favorites: [] });
-  useWatchHistoryStore.setState({ watchHistory: [] });
-  useProfileStore.getState().setProfile(profile);
+  private getSandbox = () => {
+    return useConfigStore.getState().getIntegration().useSandbox ?? true;
+  };
+
+  listProfiles = async () => {
+    if (typeof this.profileService?.listProfiles === 'undefined') {
+      throw new Error('listProfiles is not available in profile service');
+    }
```

Comment:
Not important, but a bit cleaner: perhaps we can use [safe-guards with type assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions)

```ts
// can be made a generic util function
function assertMethod<T>(method: T, message: string): asserts method is NonNullable<T> {
  if (!method) throw new Error(message);
}

class ProfileController {
  // constructor

  async listProfiles() {
    assertMethod(this.profileService?.listProfiles, 'listProfiles is not available in profile service');
    
    // `profileService.listProfiles` is concrete here
  }
}
```